sys/kern/subr_bus.c

```diff
@@ -1205,7 +1205,12 @@ static int
 devclass_alloc_unit(devclass_t dc, device_t dev, int *unitp)
 {
 	const char *s;
-	int unit = *unitp;
+	u_int unit = *unitp;
+
+	_Static_assert((int)DEVICE_UNIT_MAX > 0,
+	    "DEVICE_UNIT_MAX must be non-negative when converted to integer");
+	_Static_assert(DEVICE_UNIT_MAX < DEVICE_UNIT_ANY,
+	    "DEVICE_UNIT_MAX must be less than DEVICE_UNIT_ANY");
```

Comment:
These static asserts are misplaced.
Also, won't the second one fire since INT_MAX is not < -1?
Also, why not just use INT_MAX?

Reply:
Second doesn't trigger since `DEVICE_UNIT_MAX` was declared `unsigned`.  This seemed a reasonable place since everything was in `devclass_alloc_unit()`.